tests/moses/scoring/enum_bscoreUTest.cxxtest

```diff
@@ -0,0 +1,81 @@
+/** enum_bscoreUTest.cxxtest ---
+ *
+ * Copyright (C) 2019 OpenCog Foundation
+ *
+ * Authors: Kiros Hailay <se.kiros.hailay12@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+
+#include <cxxtest/TestSuite.h>
+#include <opencog/moses/scoring/bscores.h>
+#include <opencog/atoms/base/Link.h>
+#include <opencog/data/table/table.h>
+#include <opencog/data/table/table_io.h>
+
+#include <opencog/utils/value_key.h>
+#include <opencog/utils/valueUtils.h>
+
+using namespace opencog;
+using namespace combo;
+using namespace moses;
+using namespace std;
+
+const string utest_dir=string(PROJECT_SOURCE_DIR) + "/tests/moses/scoring/";
+
+class enum_bscoreUTest : public CxxTest::TestSuite
+{
+public:
+
+	void test_enum_bscore(void)
+	{
+		string input_file = utest_dir + "dataset2.ctable";
+		CTable ct = loadCTable(input_file);
+		enum_table_bscore et_bscore(ct);
+
+		Handle f1 = createNode(PREDICATE_NODE, "f1");
+		Handle f2 = createNode(PREDICATE_NODE, "f2");
+		Handle f3 = createNode(PREDICATE_NODE, "f3");
+		ValueSeq _f1_values, _f2_values, _f3_values;
+
+		boost::transform(ct.get_input_col_data(0), back_inserter(_f1_values),
+		                 bool_vertex_to_value);
+		boost::transform(ct.get_input_col_data(1), back_inserter(_f2_values),
+		                 bool_vertex_to_value);
+		boost::transform(ct.get_input_col_data(2), back_inserter(_f3_values),
+		                 bool_vertex_to_value);
+
+		ValuePtr f1_proto_atom(new LinkValue(_f1_values));
+		f1->setValue(moses::compressed_value_key, f1_proto_atom);
+		ValuePtr f2_proto_atom(new LinkValue(_f2_values));
+		f2->setValue(moses::compressed_value_key, f2_proto_atom);
+		ValuePtr f3_proto_atom(new LinkValue(_f3_values));
+		f3->setValue(moses::compressed_value_key, f3_proto_atom);
+
+		HandleSeq handleSeq = {f1, f2, f3};
+		Handle program = createLink(handleSeq, COND_LINK);
+
+		behavioral_score bs = et_bscore(program);
+		// note that the compressed table might get reordered in this case the
+		// fourth row will come at the first, first row will come at the second
+		// , third row will come at  the Since the population of the input data
```

Comment:
1. `,` shouldn't be at the beginning of a line.
2. "come at   the", looks like some word is missing here.